docs/deep-dive/scaffolding/your-first-scaffold-plugin.md

```diff
@@ -0,0 +1,138 @@
+    - **input** - An object containing the inputted values received from your inquirer questions
+    - **oraSpinner** - Remember the pretty spinner you see when interacting with our CLI? This is the [Ora Spinner](https://www.npmjs.com/package/ora) instance for you to use in case you need to update it.
+
+The **context** object referenced above contains various data that might be helpful: certain paths, the config file's data, registered plugins, ... . Feel free to output it & take a deeper look at it.
+
+
+# Example
+Let's create our first scaffolding plugin. I've navigated to my project's root and created a new folder: `./scaffoldFolder`. Here I'll quickly initialise a new package using `npm init -y`.
+
+I've taken the liberty of writing a short plugin in `./scaffoldFolder/index.js` that creates a customly named Folder in my project's root:
+
+```js
+const fs = require("fs");
+const path = require("path");
+
+module.exports = {
+    name: "cli-plugin-scaffold-template-scaffold-folder",
+    type: "cli-plugin-scaffold-template",
+    scaffold: {
+        name: "Scaffold folder",
+        questions: ({ context }) => {
+            return [
+                {
+                    name: "folderName",
+                    message: "What's the name of the folder you'd like to create? ",
+                    validate: value => {
+                        if (value.match(/[^a-zA-Z0-9]/)) {
+                            return "Please insert a folder name containing only alphanumeric characters";
+                        }
+
+                        const projectRootPath = context.paths.projectRoot;
+                        if (fs.existsSync(path.join(projectRootPath, value))) {
+                            return "A folder with the same name already exists in this path";
+                        }
+
+                        return true;
+                    }
+                }
+            ];
+        },
+        generate: async ({ input, context }) => {
+            const { folderName } = input;
+            const projectRootPath = context.paths.projectRoot;
+
+            fs.mkdirSync(path.join(projectRootPath, folderName));
+            fs.writeFileSync("success.txt", "Folder has been successfully scaffolded!");
+        }
+    }
+};
+```
+
+You may see that I employ **inquirer validation** in order to ensure that the folder's name contains only alphanumeric characters. We'll also check whether the folder already exists using the path supplied by **context.paths.projectRoot**.
+
+Now we're able to run `webiny scaffold` within our project's root, select our newly created scaffold plugin and fill in a folder name in order to create it. Of course, it will warn and prompt us to insert a different name if the validation fails.
+
+If you've made it so far, congratulations! You've just finished your first scaffold plugin. 🚀
```

Comment:
IMHO
`Congratulations!` instead of `If you've made it so far, congratulations!`

Reply:
done